src/main/java/com/influy/domain/faqCategory/service/FaqCategoryServiceImpl.java

```diff
@@ -57,50 +59,71 @@ public Page<FaqCategory> getPage(Long sellerId, Long itemId, PageRequestDto page
-            FaqCategory faqCategory = faqCategoryRepository.findById(request.getId())
-                    .orElseThrow(() -> new GeneralException(ErrorStatus.FAQ_CATEGORY_NOT_FOUND));
+        FaqCategory faqCategory = faqCategoryRepository.findById(request.getId())
+                .orElseThrow(() -> new GeneralException(ErrorStatus.FAQ_CATEGORY_NOT_FOUND));
+
+        item.getFaqCategoryList().remove(faqCategory);
+        faqCardRepository.deleteAllByFaqCategory(faqCategory); // 해당 faq 카테고리에 있는 모든 faq 카드 삭제
+        faqCategoryRepository.delete(faqCategory);
 
-            item.getFaqCategoryList().remove(faqCategory);
-            faqCategoryRepository.delete(faqCategory);
+        // 순서 정리
+        List<FaqCategory> faqCategoryList = faqCategoryRepository.findAllByItemId(itemId, Sort.by(Sort.Direction.ASC, "categoryOrder"));
+        for (int i = 0; i < faqCategoryList.size(); i++) {
+            faqCategoryList.get(i).setCategoryOrder(i + 1);
         }
     }
 
     @Override
     @Transactional
-    public List<FaqCategory> updateAll(Long sellerId, Long itemId, List<FaqCategoryRequestDto.UpdateDto> requestList) {
-        if (!sellerRepository.existsById(sellerId)) {
-            throw new GeneralException(ErrorStatus.SELLER_NOT_FOUND);
-        }
+    public FaqCategory update(Long sellerId, Long itemId, FaqCategoryRequestDto.UpdateDto request) {
+        if (!sellerRepository.existsById(sellerId)) throw new GeneralException(ErrorStatus.SELLER_NOT_FOUND);
+        Item item = itemRepository.findById(itemId)
+                .orElseThrow(() -> new GeneralException(ErrorStatus.ITEM_NOT_FOUND));
+
+        FaqCategory faqCategory = faqCategoryRepository.findById(request.getId())
+                .orElseThrow(() -> new GeneralException(ErrorStatus.FAQ_CATEGORY_NOT_FOUND));
+
+        if (!faqCategory.getItem().getId().equals(item.getId())) throw new GeneralException(ErrorStatus.INVALID_FAQ_ITEM_RELATION);
+
+        if (request.getCategory() != null) faqCategory.setCategory(request.getCategory());
+        faqCategoryRepository.save(faqCategory);
+
+        return faqCategory;
+    }
+
+    @Override
+    @Transactional
+    public List<FaqCategory> updateOrderAll(Long sellerId, Long itemId, FaqCategoryRequestDto.UpdateOrderDto request) {
+        if (!sellerRepository.existsById(sellerId)) throw new GeneralException(ErrorStatus.SELLER_NOT_FOUND);
 
         Item item = itemRepository.findById(itemId)
                 .orElseThrow(() -> new GeneralException(ErrorStatus.ITEM_NOT_FOUND));
 
+        List<Long> idList= request.getIds();
         List<FaqCategory> updatedList = new ArrayList<>();
 
-        for (FaqCategoryRequestDto.UpdateDto request : requestList) {
-            FaqCategory faqCategory = faqCategoryRepository.findById(request.getId())
+        for (int i=0; i<idList.size(); i++) {
+            Long id = idList.get(i);
+            FaqCategory faqCategory = faqCategoryRepository.findById(id)
```

Comment:
이러면 반복문에서 i 증가할때마다 조회 쿼리가 나가게 될 것 같아서 혹시 이런 방법은 어떨까요?!

```suggestion
            Map<Long, FaqCategory> map = new HashMap<>();
            
            faqCategoryRepository.findAllByItem(item).stream()
                 .map(category->map.put(category.getId(),category))...//쿼리는 한번만
            
            for (int i=0; i<idList.size(); i++) {
                 Long id = idList.get(i);
                 FaqCategory faqCategory = map.get(id);
                 if((faqCategory.getCategoryOrder()).equals(i+1){
                      continue;     //순서 변경사항이 없으면 수정 쿼리 날리지 않기
                  }
                 faqCategory.setCategoryOrder(i+1);
            }
```

Reply:
오 미리 받는게 좋을 것 같네요! 그렇게 리팩토링 완료 했슴당 :)